File: K-th element of two Arrays.java
class Solution {
    public int kthElement(int a[], int b[], int k) {
        // code here
        int i ;
        int n = a.length;
        int m = b.length;
       int[] arr = new int[n+m];
        for( i = 0;i<n;++i) arr[i] = a[i];
        int j = 0;
        for(  ; i<n+m;++i){
            arr[i]=b[j];
            ++j;
        }
        Arrays.sort(arr);
        return arr[k-1];
    }
}

File: K-th element of two Arrays.py
class Solution:
    def kthElement(self, a, b, k):
        # code here
        if len(a) > len(b):
            return self.kthElement( b, a, k)
        n, m = len(a), len(b)
        low = max(0, k - m)
        high = min(k, n)
    
        while low <= high:
            cutA = (low + high) // 2
            cutB = k - cutA
    
            leftA = a[cutA - 1] if cutA > 0 else float('-inf')
            leftB = b[cutB - 1] if cutB > 0 else float('-inf')
            rightA = a[cutA] if cutA < n else float('inf')
            rightB = b[cutB] if cutB < m else float('inf')
    
            if leftA <= rightB and leftB <= rightA:
                return max(leftA, leftB)
            elif leftA > rightB:
                high = cutA - 1
            else:
                low = cutA + 1
    
        return -1
